### PR TITLE
Implement Attribute Truncation in to_cosmosdb_item to Comply with CosmosDB Size Limit

### PR DESCRIPTION
**Pull request by liucheng-ms**
# Description

This PR add the `to_cosmosdb_item` method in the `Span` class, ensuring that attribute values are truncated to a specified maximum length. This change is critical to prevent exceeding CosmosDB's 2MB item size limit. The `attr_value_truncation_length` parameter allows for flexible adjustment of the truncation threshold, accommodating different use cases and requirements.

By merging this PR, we will improve the reliability of our telemetry data storage in CosmosDB, especially for larger spans that previously risked rejection due to size limitations. The code aligns with the serialization method used by the CosmosDB Python SDK, ensuring compatibility and consistency across our application.

# All Promptflow Contribution checklist:
- [X] **The pull request does not introduce [breaking changes].**
- [X] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [X] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [X] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [X] Title of the pull request is clear and informative.
- [X] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [X] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: src/promptflow-azure/promptflow/azure/_storage/cosmosdb/span.py [hunk 1]
```diff
@@ -65,13 +65,27 @@ def persist(self, cosmos_client: ContainerProxy, blob_container_client: Containe
...
+    def to_cosmosdb_item(self, max_attr_value_length: int = 2048):
+        """
+        Convert the object to a dictionary for persistence to CosmosDB.
+        Truncate attribute values to avoid exceeding CosmosDB's 2MB size limit.
+        """
+        item = self.to_dict()  # use to_dict method to get a dictionary representation of the object
+        attributes = item.get("attributes")
+        if attributes:
+            item["attributes"] = {
+                k: (v if not isinstance(v, str) or len(v) <= max_attr_value_length else v[:max_attr_value_length])
```
Comment (thy09) [inline src/promptflow-azure/promptflow/azure/_storage/cosmosdb/span.py:84]: no need to check `len(v) <= max_attr_value_lengeh`

**Review by thy09 — APPROVED** [review]

File: src/promptflow-azure/promptflow/azure/_storage/cosmosdb/span.py (hunk 1 above)
Comment (liucheng-ms) [reply src/promptflow-azure/promptflow/azure/_storage/cosmosdb/span.py]: Sure.

File: src/promptflow-azure/promptflow/azure/_storage/cosmosdb/span.py [hunk 2]
```diff
@@ -65,13 +65,36 @@ def persist(self, cosmos_client: ContainerProxy, blob_container_client: Containe
...
+        item = self.to_dict()  # use to_dict method to get a dictionary representation of the object
+
+        # Serialize the dictionary to a JSON string using separators (",", ":").
+        # This is done to align with the serialization method used by the CosmosDB Python SDK.
+        json_dumped = json.dumps(item, separators=(",", ":"))
+
+        item_size = len(json_dumped.encode("utf-8"))
+        max_size_in_bytes = 2 * 1024 * 1024  # 2MB in bytes
+
+        if item_size > max_size_in_bytes:
```
Comment (zzn2) [inline src/promptflow-azure/promptflow/azure/_storage/cosmosdb/span.py:89]: Add a comment to say that the cosmos db's limit is 2MB, and we do truncations for long attributes here.

File: src/promptflow-azure/promptflow/azure/_storage/cosmosdb/span.py [hunk 3]
```diff
@@ -65,13 +65,36 @@ def persist(self, cosmos_client: ContainerProxy, blob_container_client: Containe
...
+            attributes = item.get("attributes")
+            if attributes:
+                item["attributes"] = {
+                    k: (v if not isinstance(v, str) else v[:attr_value_truncation_length])
```
Comment (zzn2) [inline src/promptflow-azure/promptflow/azure/_storage/cosmosdb/span.py:93]: How about
```suggestion
                    k: (v[:attr_value_truncation_length] if isinstance(v, str) else v)
```

File: src/promptflow/tests/sdk_cli_azure_test/unittests/test_span.py [hunk 4]
```diff
@@ -92,6 +93,82 @@ def test_to_dict(self):
...
+                },
+                kind="test",
+                parent_id="test",
+                start_time=datetime.datetime.fromisoformat("2022-01-01T00:00:00"),
+                end_time=datetime.datetime.fromisoformat("2022-01-01T00:01:00"),
+                status={},
+                attributes={
+                    "attr1": "a" * 1024 * 1024,  # 1MB
+                    "attr2": "b" * 1024 * 1024,  # 1MB
+                    "attr3": "c" * 1024 * 1024,  # 1MB
```
Comment (zzn2) [inline src/promptflow/tests/sdk_cli_azure_test/unittests/test_span.py:114]: Let's add non-str values here to test whether non-str values are not truncated.

File: src/promptflow-azure/promptflow/azure/_storage/cosmosdb/span.py (hunk 2 above)
Comment (liucheng-ms) [reply src/promptflow-azure/promptflow/azure/_storage/cosmosdb/span.py]: Will remove size condition for clear and fixed behavior for customer.

**Review by zhengfeiwang — APPROVED** [review]

**Review by thy09 — APPROVED** [review]

**Pull request merged**